### PR TITLE
Fengwu ghr

### PR DESCRIPTION
**Pull request by rnwzd**

### CODE REVIEW

File: graph_weather/models/fengwu_ghr/layers.py [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    y = y / den
+
+    return y
+
+
+def grid_interpolate(lat_lons: list, z: torch.Tensor,
+                     height, width,
+                     method: str = "cubic"):
+    # TODO 1. CPU only
+    #      2. The mesh is a rectangle, not a sphere
```
Comment (jacobbieker) [inline graph_weather/models/fengwu_ghr/layers.py:44]: It being a rectangle is fine, it has to be for the MetaModel to take it as an image. So I think this is great!

File: graph_weather/models/fengwu_ghr/layers.py [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+                    k: int = 3, num_workers: int = 1):
+    with torch.no_grad():
+        assign_index = knn(pos_x, pos_y, k,
+                           num_workers=num_workers)
+        y_idx, x_idx = assign_index[0], assign_index[1]
+        diff = pos_x[x_idx] - pos_y[y_idx]
+        squared_distance = (diff * diff).sum(dim=-1, keepdim=True)
+        weights = 1.0 / torch.clamp(squared_distance, min=1e-16)
+
+    # print((x[x_idx]*weights).shape)
```
Comment (jacobbieker) [inline graph_weather/models/fengwu_ghr/layers.py:29]: I would remove commented out code like this when you finish with the changes, just to keep it a bit cleaner.

**Review by jacobbieker — COMMENTED** [review]
Looks great so far! Just a few comments.

**Comment by peterdudfield** [comment]
@all-contributors please add @rnwzd  for code

**Pull request merged**